azuredevops/internal/service/feed/resource_feed_permission.go

```diff
@@ -0,0 +1,254 @@
+package feed
+
+import (
+	"fmt"
+	"net/http"
+
+	"github.com/google/uuid"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+	"github.com/microsoft/azure-devops-go-api/azuredevops/v7"
+	"github.com/microsoft/azure-devops-go-api/azuredevops/v7/feed"
+	"github.com/microsoft/azure-devops-go-api/azuredevops/v7/graph"
+	"github.com/microsoft/azure-devops-go-api/azuredevops/v7/identity"
+	"github.com/microsoft/terraform-provider-azuredevops/azuredevops/internal/client"
+	"github.com/microsoft/terraform-provider-azuredevops/azuredevops/internal/utils"
+	"github.com/microsoft/terraform-provider-azuredevops/azuredevops/internal/utils/converter"
+)
+
+func ResourceFeedPermission() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceFeedPermissionCreate,
+		Read:   resourceFeedPermissionRead,
+		Update: resourceFeedPermissionUpdate,
+		Delete: resourceFeedPermissionDelete,
+		Schema: map[string]*schema.Schema{
+			"feed_id": {
+				Type:         schema.TypeString,
+				ValidateFunc: validation.IsUUID,
+				Required:     true,
+				ForceNew:     true,
+			},
+			"identity_descriptor": {
+				Type:         schema.TypeString,
+				ValidateFunc: validation.StringIsNotWhiteSpace,
+				Required:     true,
+				ForceNew:     true,
+			},
+			"identity_id": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"role": {
+				Type: schema.TypeString,
+				ValidateFunc: validation.StringInSlice([]string{
+					string(feed.FeedRoleValues.Reader),
+					string(feed.FeedRoleValues.Contributor),
+					string(feed.FeedRoleValues.Administrator),
+					string(feed.FeedRoleValues.Collaborator),
+				}, false),
+				Required: true,
+			},
+			"project_id": {
+				Type:         schema.TypeString,
+				ValidateFunc: validation.IsUUID,
+				Optional:     true,
+				ForceNew:     true,
+			},
+			"display_name": {
+				Type:         schema.TypeString,
+				ValidateFunc: validation.StringIsNotWhiteSpace,
+				Optional:     true,
+			},
+		},
+	}
+}
+
+func resourceFeedPermissionCreate(d *schema.ResourceData, m interface{}) error {
+	clients := m.(*client.AggregatedClient)
+
+	feed_id := d.Get("feed_id").(string)
```

Comment:
The naming style is camel case style, 
```suggestion
	feedId := d.Get("feed_id").(string)
```